pkg/gen/node-templates/provider.ts.mustache

```diff
@@ -1,9 +1,110 @@
 // *** WARNING: this file was generated by the Pulumi Kubernetes codegen tool. ***
 // *** Do not edit by hand unless you're certain you know what you are doing! ***
 
+import * as fs from "fs";
+
 import * as pulumi from "@pulumi/pulumi";
 import * as inputApi from "./types/input";
 import * as outputApi from "./types/output";
+import * as yaml from "js-yaml";
+import * as glob from "glob";
+
+export interface YamlOpts {
+    files?: string[] | string;
+    yaml?: string[] | string;
+    transformations?: ((o: any) => void)[];
+}
+
+/**
+ * Yaml creates a set of Kubernetes resources from Kubernetes YAML text. The YAML text may be
+ * supplied using any of the following `YamlOpts`:
+ *
+ *   1. Using a filename or a list of filenames:
+ *        a. `{files: "foo.yaml"}`
+ *        b. `{files: ["foo.yaml", "bar.yaml"]}`
+ *   2. Using a file pattern or a list of file patterns:
+ *        a. `{files: "*.yaml"}`
+ *        b. `{files: ["foo/*.yaml", "bar/*.yaml"]}`
+ *   3. Using a literal string containing YAML, or a list of such strings:
+ *        a. `{yaml: "(LITERAL YAML HERE)"}`
+ *        b. `{yaml: ["(LITERAL YAML HERE)", "(MORE YAML)"]}`
+ *   4. Any combination of files, patterns, or YAML strings:
+ *        a. `{files: "foo.yaml", yaml: "(LITERAL YAML HERE)"}`
+ */
+export class Yaml {
```

Comment:
Can you go over the thinking behind making this a plain class rather than inheriting from `pulumi.ComponentResource`? I do think it makes sense--e.g. it allows the use of the type without introducing potentially-awkward nesting--but I'd like to hear your thoughts as well.

Reply:
Definitely would go the component route so stuff gets parented nicely underneath each yaml template.

Reply:
:+1: This should be a component, and should parent the resources it creates.

We do have to consider whether the Helm Chart component has as children its direct Kube resources or a set of these Yaml components (whcih themselves have kube resources as children).

Reply:
I didn't think it added much value and we needed the YAML objects to be parented to the `helm.Chart` in that case. But it sounds like people like this idea so we can do that.